sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -406,6 +406,150 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
             raise ValueError("Unsupported BlobType: {}".format(blob_type))
         return kwargs
 
+    def _upload_blob_from_url_options(self, source_url, **kwargs):
+        # type: (...) -> Dict[str, Any]
+        headers = kwargs.pop('headers', {})
+        tier = kwargs.pop('standard_blob_tier', None)
+        include_source_blob_properties = kwargs.pop('include_source_blob_properties', True)
+        timeout = kwargs.pop('timeout', None)
+        dest_mod_conditions = get_modify_conditions(kwargs)
+        blob_tags_string = serialize_blob_tags_header(kwargs.pop('tags', None))
+        content_settings = kwargs.pop('content_settings', None)
+        source_content_md5 = kwargs.pop('source_content_md5', None)
+        if content_settings:
+            kwargs['blob_http_headers'] = BlobHTTPHeaders(
+                blob_cache_control=content_settings.cache_control,
+                blob_content_type=content_settings.content_type,
+                blob_content_md5=None,
+                blob_content_encoding=content_settings.content_encoding,
+                blob_content_language=content_settings.content_language,
+                blob_content_disposition=content_settings.content_disposition
+            )
+        cpk = kwargs.pop('cpk', None)
+        cpk_info = None
+        if cpk:
+            if self.scheme.lower() != 'https':
+                raise ValueError("Customer provided encryption key must be used over HTTPS.")
+            cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
+                               encryption_algorithm=cpk.algorithm)
+        kwargs['cpk_info'] = cpk_info
+        kwargs['cpk_scope_info'] = get_cpk_scope_info(kwargs)
+
+        options = {
+            'content_length': 0,
+            'copy_source_blob_properties': include_source_blob_properties,
+            'source_content_md5': bytearray(source_content_md5) if source_content_md5 else None,
+            'copy_source': source_url,
+            'timeout': timeout,
+            'modified_access_conditions': dest_mod_conditions,
+            'blob_tags_string': blob_tags_string,
+            'headers': headers,
+            'cls': return_response_headers,
+        }
+        source_mod_conditions = get_source_conditions(kwargs)
+        dest_access_conditions = get_access_conditions(kwargs.pop('destination_lease', None))
+        options['source_modified_access_conditions'] = source_mod_conditions
+        options['lease_access_conditions'] = dest_access_conditions
+        options['tier'] = tier.value if tier else None
+        options.update(kwargs)
+        return options
+
+    @distributed_trace
+    def upload_blob_from_url(self, source_url, **kwargs):
```

Comment:
just want to confirm again if we want to have `overwrite` parameter

Reply:
It is not supported, I have checked with .NET as well. It will just overwrite by default.

Reply:
in arch board meeting @kasobol-msft mentioned we want make it like upload_blob, also it seems we have overwirte param in java @rickle-msft, can you help comfirm this

Reply:
I have just checked looks like it is in fact supported in Java, looking into adding this for Python and .Net @seanmcc-msft?